packages/santa_close_design-system/src/utils/generateVariants.ts

```diff
@@ -0,0 +1,22 @@
+import {CSSProperties} from '@stitches/react'
+import {Theme, theme} from '../styles'
+
+const createMultipleCSSProp = (cssProp: (keyof CSSProperties)[], value) =>
+  Object.fromEntries(cssProp.map((prop) => [prop, value]))
+
+export const generateVariants = <T extends keyof Theme>(
+  cssProp: keyof CSSProperties | (keyof CSSProperties)[],
+  target: T,
+) => {
+  const isMultiple = Array.isArray(cssProp) && cssProp.length > 1
+
+  return Object.fromEntries(
+    Object.entries(theme[target]).map(([key, value]) => {
+      const returnValue = isMultiple
+        ? createMultipleCSSProp(cssProp, value)
+        : {[cssProp as string]: value}
+
+      return [key, returnValue]
+    }),
+  )
```

Comment:
```suggestion
  ) as {
    [ThemeToken in keyof Theme[T]]: {
      [TokenVariants in keyof Theme[T][ThemeToken]]: Theme[T][ThemeToken][TokenVariants]
    }
  }
```
요거 추가해주면 타입추론 조금 더 잘 될것 같긴한데 `createMultipleCSSProp`함수가 어떤 동작해주는건지 정확하게 모르겠네연 ㅎ

Reply:
타입추론과 오류를 줄이기 위해 선언 방식으로 처리했습니다~